internal/http/services/overleaf/overleaf.go

```diff
@@ -0,0 +1,293 @@
+}
+
+func init() {
+	global.Register("overleaf", New)
+}
+
+func New(ctx context.Context, m map[string]interface{}) (global.Service, error) {
+	conf := &config{}
+	if err := mapstructure.Decode(m, conf); err != nil {
+		return nil, err
+	}
+
+	conf.init()
+
+	gtw, err := pool.GetGatewayServiceClient(pool.Endpoint(conf.GatewaySvc))
+	if err != nil {
+		return nil, err
+	}
+
+	r := chi.NewRouter()
+
+	s := &svc{
+		conf:      conf,
+		gtwClient: gtw,
+		router:    r,
+	}
+
+	if err := s.routerInit(); err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func (s *svc) routerInit() error {
+	s.router.Get("/import", s.handleImport)
+	s.router.Post("/export", s.handleExport)
+	return nil
+}
+
+func (c *config) init() {
+	if c.Prefix == "" {
+		c.Prefix = "overleaf"
+	}
+
+	c.GatewaySvc = sharedconf.GetGatewaySVC(c.GatewaySvc)
+}
+
+// Close performs cleanup.
+func (s *svc) Close() error {
+	return nil
+}
+
+func (s *svc) Prefix() string {
+	return s.conf.Prefix
+}
+
+func (s *svc) Unprotected() []string {
+	return nil
+}
+
+func (s *svc) Handler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		s.router.ServeHTTP(w, r)
+	})
+}
+
+func (s *svc) handleImport(w http.ResponseWriter, r *http.Request) {
+	reqres.WriteError(w, r, reqres.APIErrorUnimplemented, "Overleaf import not yet supported", nil)
+}
+
+func (s *svc) handleExport(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := appctx.GetLogger(ctx)
+
+	statRes, err := s.validateQuery(w, r, ctx)
+	if err != nil {
+		// Validate query handles errors
+		return
+	}
+
+	resource := statRes.Info
+
+	// User needs to have download rights to export to Overleaf
+	if !resource.PermissionSet.InitiateFileDownload {
+		reqres.WriteError(w, r, reqres.APIErrorUnauthenticated, "permission denied when accessing the file", err)
+		return
+	}
+
+	if resource.Type != storagepb.ResourceType_RESOURCE_TYPE_FILE && resource.Type != storagepb.ResourceType_RESOURCE_TYPE_CONTAINER {
+		reqres.WriteError(w, r, reqres.APIErrorInvalidParameter, "invalid resource type, resource should be a file or a folder", nil)
+		return
+	}
+
+	token, ok := ctxpkg.ContextGetToken(ctx)
+	if !ok || token == "" {
+		reqres.WriteError(w, r, reqres.APIErrorUnauthenticated, "Access token is invalid or empty", err)
+		return
+	}
+
+	if r.Form.Get("override") == "" {
+		creationTime, alreadySet := resource.GetArbitraryMetadata().Metadata["reva.overleaf.exporttime"]
+		if alreadySet {
+			w.WriteHeader(http.StatusConflict)
+			if err := json.NewEncoder(w).Encode(map[string]any{
+				"code":        "ALREADY_EXISTS",
+				"message":     "Project was already exported",
+				"export_time": creationTime,
+			}); err != nil {
+				reqres.WriteError(w, r, reqres.APIErrorServerError, "error marshalling JSON response", err)
+				return
+			}
+			w.Header().Set("Content-Type", "application/json")
+			return
+		}
+	}
+
+	// TODO: generate and use a more restricted token
+	restrictedToken := token
```

Comment:
For generating a restricted token, use a similar code to [this](https://github.com/cs3org/reva/blob/5bd1923565619e21f51728019456da010e1d5689/pkg/share/cache/warmup/cbox/cbox.go#L97-L119).

For the scope, you can use [this](https://github.com/cs3org/reva/blob/5bd1923565619e21f51728019456da010e1d5689/pkg/auth/scope/resourceinfo.go#L125C21-L125C21) function to get a restricted scope for the resource you want to archive. As role set it to viewer, and the initial scope as nil.